plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -397,4 +398,19 @@ public static Transaction newCreateTableTransaction(TrinoCatalog catalog, Connec
 
         return catalog.newCreateTableTransaction(session, schemaTableName, schema, partitionSpec, targetPath, propertiesBuilder.build());
     }
+
+    public static long getSnapshotIdAsOfTime(Table table, long timestampMillis)
+    {
+        Long snapshotId = null;
+        for (HistoryEntry logEntry : table.history()) {
```

Comment:
This is modified based on https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java#L265-L276. The right thing to do in the long term is likely to directly call that method. I am making a copy here because that method is not released yet.

Reply:
Maybe just add a TODO comment to remind us to remove this method once we're using an Iceberg library version we can call instead?